### PR TITLE
gitInitialized needs to be set if repo is already initialized

### PR DESCRIPTION
**Pull request by brunnels**
-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -434,6 +434,12 @@ module.exports = class extends BaseGenerator {
                                         this.gitInitialized = true;
                                     });
                                 }
+
+                                // gitInitialized needs to be set if already initialized
+                                // gitDir has a line break to remove (at least on windows)
```
Comment (deepu105) [inline generators/app/index.js]: why not change the if above to `if (!gitDir.trim())` and do this in else, that looks cleaner IMO

Comment (brunnels) [reply generators/app/index.js]: Fair enough.  I was just trying to be clear what was happening.

File: generators/app/index.js [hunk 2]
```diff
@@ -428,7 +428,10 @@ module.exports = class extends BaseGenerator {
                     this.isGitInstalled(code => {
                         if (code === 0) {
                             this.gitExec('rev-parse --is-inside-work-tree', { trace: false }, (err, gitDir) => {
-                                if (!gitDir) {
+                                // gitDir has a line break to remove (at least on windows)
+                                if(gitDir && gitDir.trim() === 'true') {
```
Comment (deepu105) [inline generators/app/index.js]: is the value normally a text `'true'`?

Comment (brunnels) [reply generators/app/index.js]: When not in a git repo
```
> git rev-parse --is-inside-work-tree
fatal: Not a git repository (or any of the parent directories): .git
```
when in a git repo
```
> git rev-parse --is-inside-work-tree
true
```

Comment (deepu105) [inline generators/app/index.js]: please fix the prettier error, just run `npm run lint --fix`

**Pull request merged**